src/plugins/Miner.cpp

```diff
@@ -38,6 +39,12 @@ void SecureMineralsIncome(Builder* builder_) {
 
         builder_->ScheduleOptionalOrder(gHub->GetCurrentWorkerType(), i);
     }
+
+    // Auto-Expand if TownHalls getting full and floating too many minerals
+    if (town_halls().size() * 15 < numTotalWorkers
```

Comment:
What is this 15 here? I think we should count real number of workers, gathering something, otherwise you may build like 5 expansions in a row without adding new workers.
Also, auto-expand could be bad for some very aggressive strategies. Perhaps, we can postpone this part?

Reply:
Well, in general I don't like this hardcode as it strictly enforces particular play style, i.e. we play macro and macro and saturate as max as possible, while previous logic didn't enforce anything and triggered saturation only when a new expansions appeared. And the final decision was taken by particular strategy (i.e. if we want to play with 2 bases, we schedule a townhall and miner saturates it, we are happy with worker rash, we don't schedule anything). I think we shouldn't go this way.

So, I suggest to introduce the settings part first and revisit this patch later. The suggested behaviour can be configured via the settings, but looks more like part of particular strategy to me.

Reply:
Ok, I can postpone this until after Strategy Settings is done which can specific macro settings.